src/main/java/org/opentripplanner/routing/edgetype/StreetElevationExtension.java

```diff
@@ -80,22 +87,45 @@ public PackedCoordinateSequence getElevationProfile() {
     }
   }
 
+  /**
+   * The distance multiplied by the {@link StreetEdge#bicycleSafetyFactor}, but also considering the
+   * increased length caused by the slope.
+   */
   public double getEffectiveBicycleSafetyDistance() {
     return effectiveBicycleSafetyDistance;
   }
 
+  /**
+   * The distance multiplied by a factor considering how much faster/slower it is to bile the edge,
+   * compared to if it was flat.
+   */
   public double getEffectiveBikeDistance() {
     return effectiveBikeDistance;
   }
 
+  /**
+   * The distance multiplied by a factor considering how much more/less convenient it is to bike
+   * the edge, compared to if it was flat. This is calculated form the energy usage of the cyclist.
+   */
```

Comment:
> The distance multiplied

It it distance or cost?

Reply:
It is distance. It is divided by the speed to get the cost.

Reply:
The name for this method is misleading, as the unit is not seconds. This is used to calculate a part of the weight with the following expression , where `speed` is the request scoped speed. If weight has the unit of second, and speed m/s, unit analysis gives out that this value has the unit of meter. 

```
weight = getEffectiveBikeWorkCost() / speed;
```

Reply:
On the other hand, when calculating `effectiveBikeWorkFactor`, it says that the unit is `The slope work factor in joules per meters at 5 m/s`, which would mean that the unit is actually Joules. I really don't know how the handling of this should be done at varying speeds.

Reply:
I love it when the expected unit is `meters` and the actual output is `1/5 Joule` 🙈 . Would it be better to rename the method to `getEffectiveBikeDistanceForWorkCost` (In `StreetEdge` and `StreetElevationExtension`), and then leave it for now.